opensearch/src/main/java/org/opensearch/sql/opensearch/data/utils/OpenSearchJsonContent.java

```diff
@@ -126,28 +126,35 @@ public Object objectValue() {
   @Override
   public Pair<Double, Double> geoValue() {
     final JsonNode value = value();
-    if (value.has("lat") && value.has("lon")) {
-      Double lat = 0d;
-      Double lon = 0d;
+    Double lat = null;
+    Double lon = null;
+
+    if (value.has("lat")) {
       try {
         lat = extractDoubleValue(value.get("lat"));
       } catch (Exception exception) {
         throw new IllegalStateException(
             "latitude must be number value, but got value: " + value.get(
                 "lat"));
       }
+    }
+
+    if (value.has("lon")) {
       try {
         lon = extractDoubleValue(value.get("lon"));
       } catch (Exception exception) {
         throw new IllegalStateException(
             "longitude must be number value, but got value: " + value.get(
                 "lon"));
       }
-      return Pair.of(lat, lon);
-    } else {
-      throw new IllegalStateException("geo point must in format of {\"lat\": number, \"lon\": "
+    }
+
+    if (lat == null && lon == null) {
```

Comment:
Use `||`

Reply:
This has to be `&&`. If it's `||`, it will throw the exception for when the query is `geopoint.lat` as it only contains `lat` and `lon` would be `null`.

Reply:
maybe add a comment

Reply:
Added, thanks